ADR/ADR044-translate-api-response-at-request-time.md

```diff
@@ -0,0 +1,336 @@
+  "privacy_policy_url": null,
+  "form_slug": "a-form-for-testing-file-upload",
+  "support_email": null,
+  "support_phone": null,
+  "support_url": null,
+  "support_url_text": null,
+  "declaration_text": null,
+  "question_section_completed": false,
+  "declaration_section_completed": false,
+  "created_at": "2025-03-27T12:05:49.395300Z",
+  "updated_at": "2025-07-29T14:35:44.423406Z",
+  "creator_id": 1,
+  "what_happens_next_markdown": null,
+  "payment_url": null,
+  "submission_type": "email",
+  "share_preview_completed": false,
+  "s3_bucket_name": null,
+  "s3_bucket_aws_account_id": null,
+  "s3_bucket_region": null,
+  "language": "en",
+  "start_page": 33,
+  "steps": [
+    {
+      "id": 33,
+      "position": 1,
+      "next_step_id": null,
+      "type": "question_page",
+      "data": {
+        "question_text": "Upload your file",
+        "hint_text": "",
+        "answer_type": "file",
+        "is_optional": false,
+        "answer_settings": null,
+        "page_heading": "Annual report",
+        "guidance_markdown": "Some guidance",
+        "is_repeatable": false
+      },
+      "routing_conditions": []
+    }
+  ]
+}
+```
+
+And `/forms/7/live?locale=cy`, would return the same object with any Welsh translations overwriting the English:
+```jsonc
+{
+  "form_id": "7",
+  "name": "Ffurflen ar gyfer profi uwchlwytho ffeiliau",
+  "submission_email": "david.biddle@digital.cabinet-office.gov.uk",
+  "privacy_policy_url": null,
+  "form_slug": "a-form-for-testing-file-upload",
+  "support_email": null,
+  "support_phone": null,
+  "support_url": null,
+  "support_url_text": null,
+  "declaration_text": null,
+  "question_section_completed": false,
+  "declaration_section_completed": false,
+  "created_at": "2025-03-27T12:05:49.395300Z",
+  "updated_at": "2025-07-29T14:35:44.423406Z",
+  "creator_id": 1,
+  "what_happens_next_markdown": null,
+  "payment_url": null,
+  "submission_type": "email",
+  "share_preview_completed": false,
+  "s3_bucket_name": null,
+  "s3_bucket_aws_account_id": null,
+  "s3_bucket_region": null,
+  "language": "en",
+  "start_page": 33,
+  "steps": [
+    {
+      "id": 33,
+      "position": 1,
+      "next_step_id": null,
+      "type": "question_page",
+      "data": {
+        "question_text": "Llwythwch eich ffeil i fyny",
+        "hint_text": "",
+        "answer_type": "file",
+        "is_optional": false,
+        "answer_settings": null,
+        "page_heading": "Adroddiad blynyddol",
+        "guidance_markdown": "Peth arweiniad",
+        "is_repeatable": false
+      },
+      "routing_conditions": []
+    }
+  ]
+}
+```
+
+Advantages:
+- Request size doesn't increase
+- API structure doesn't change aside from the addition of a query parameter.
+- Keeps the translation logic out of runner
+Disadvantages:
+- Requires more logic in the API to choose the right translation - we already have some conversion logic, but we're currently trying to reduce the amount of logic in this repo.
```

Comment:
How would we implement storing the multiple locales in the form_documents in form-api?

Reply:
My thinking was that we'd store it in a format analogous to the ones proposed in option one. 

I imagine storing them as side-by-side suffixed columns would be easiest:

```json
"name_en": "A form for testing file upload",
"name_cy": "Ffurflen ar gyfer profi uwchlwytho ffeiliau",
```

but I wasn't sure if this was worth including in the ADR or if we could leave it up to whoever picks up the implementation to choose an appropriate approach

Reply:
Does this ADR also supersede https://github.com/alphagov/forms/blob/main/ADR/ADR043-use-mobility-with-column-strategy.md then?

Reply:
Storing separate form documents for each language is essentially option 3. 

I've updated the ADR to recommend that option instead - I think it's a better fit given a) the concerns around essentially maintaining two separate schemas and b) plans to merge forms-api and forms-admin have become more concrete since I did the spike this ADR is based on